flink-staging/flink-gelly/src/main/java/org/apache/flink/graph/example/GraphMetrics.java

```diff
@@ -63,17 +63,17 @@ public static void main(String[] args) throws Exception {
 				.getRandomEdges(env, NUM_VERTICES), env);
 		
 		/** get the number of vertices **/
-		DataSet<Integer> numVertices = graph.numberOfVertices();
+		long numVertices = graph.numberOfVertices();
 		
 		/** get the number of edges **/
-		DataSet<Integer> numEdges = graph.numberOfEdges();
+		long numEdges = graph.numberOfEdges();
 		
 		/** compute the average node degree **/
 		DataSet<Tuple2<Long, Long>> verticesWithDegrees = graph.getDegrees();
 
 		DataSet<Double> avgNodeDegree = verticesWithDegrees
 				.aggregate(Aggregations.SUM, 1).map(new AvgNodeDegreeMapper())
-				.withBroadcastSet(numVertices, "numberOfVertices");
+				.withBroadcastSet(env.fromElements(numVertices), "numberOfVertices");
```

Comment:
Since `numVertices` is a single value, you can just pass it as an argument to `AvgNodeDegreeMapper()`. No need for a broadcast variable.